services/frontend-service/src/ui/ActionsCart/CheckoutDialog.tsx

```diff
@@ -78,11 +74,66 @@ const ApplyButton = (props: { openNotification: (msg: string) => void; closeDial
     }
 };
 
+const lockActions = [ActionTypes.CreateEnvironmentLock, ActionTypes.CreateApplicationLock] as const;
+type lockAction = typeof lockActions[number];
+const hasLockAction = (actions: BatchAction[]) =>
+    actions.some((act) => lockActions.includes(GetActionDetails(act).type as lockAction));
+
+const updateMessage = (act: BatchAction, m: string): BatchAction => {
```

Comment:
:arrow_up: this should probably not have the same input and output types.

Reply:
I tried to do this, I added new types for the lock details with no message. and then a general CartAction type as a superposition of Batch Action and the Lock details. I think this is better than creating new types for every action. but also I'm not totally sure about it. I still need to fix the tests. but I thought to check first.